unlock-app/src/components/interface/locks/Settings/forms/UpdateReferralFee.tsx

```diff
@@ -16,108 +24,170 @@ interface UpdateReferralFeeProps {
-          title="Referrer fee %"
-          enabled={referralFee}
-          disabled={disabledInput}
-          setEnabled={(enabled) => {
-            setReferralFee(enabled)
-            setValue(
-              'referralFeePercentage',
-              enabled ? (referralFeePercentage ?? 0) / 100 : 0
-            )
-          }}
-        />
         <Input
           type="number"
           {...register('referralFeePercentage', {
             valueAsNumber: true,
-            min: 0,
+            min: 1,
             max: 100,
           })}
           error={
             errors?.referralFeePercentage &&
-            'This field accept percentage value between 0 and 100.'
+            'This field accept percentage value between 1 and 100.'
           }
-          disabled={disabledInput || !referralFee}
+          disabled={isDisabledReferrerInput}
+        />
+      </div>
+
+      <div className="grid gap-2">
+        <ToggleSwitch
+          title="Referrer address"
+          enabled={isReferralAddressEnabled}
+          disabled={isDisabledReferrerInput}
+          setEnabled={(enabled: boolean) => {
+            setIsReferralAddressEnabled(enabled)
+          }}
+        />
+
+        <AddressInput
+          label=""
+          value={referralAddress}
+          disabled={isDisabledReferrerInput || !isReferralAddressEnabled}
+          onChange={(value: any) => {
+            setValue('referralAddress', value)
+          }}
+          error={errors?.referralAddress?.message}
+          onResolveName={onResolveName}
         />
       </div>
+
       {isManager && (
         <Button
           className="w-full md:w-1/3"
           type="submit"
-          disabled={disabledInput}
+          disabled={
+            isDisabledReferrerInput || !referralFeePercentage || !isValidAddress
+          }
           loading={setReferrerFeeMutation.isLoading}
         >
           Apply
         </Button>
       )}
+
+      <div className="flex flex-col mt-5 items-start">
+        <Card.Title>Referrers ({referralFees.length})</Card.Title>
+
+        {isLoading ? (
+          <div className="w-full flex items-center justify-center">
+            <LoadingIcon />
+          </div>
+        ) : referralFees.length ? (
+          <div className="w-full grid gap-5 grid-cols-1 mt-3">
+            {referralFees.map(({ id, referrer, fee }) => {
+              const feeNumber = Number(fee)
+
+              return (
+                <div
+                  key={id}
+                  className="w-full overflow-x-auto flex justify-between items-center border border-gray-300 p-3 rounded-xl"
+                >
+                  <h2 className="text-md font-medium break-words">
+                    {referrer}
+                  </h2>
+                  <h2 className="text-md pl-3 font-medium break-words">
+                    ({feeNumber / 100}%)
+                  </h2>
+                </div>
+              )
+            })}
+          </div>
```

Comment:
I wanted to put this in `@unlock-protocol/ui` but i guess it should be added if it's a major component used throughout the UI ? Let me know if this is fine or if i should put this in `@unlock-protocol/ui` 👍

Reply:
The way we add components in the UI library is if something is repeated at least a couple times in the main codebase. 

I think it's fine to not worry about it for now.

Reply:
Sounds good🙏